### PR TITLE
Add evaluation_loss to the estimator base class.

### PR DESCRIPTION
**Pull request by liuzh91**
## Description ##
[Bug_fix] Add evaluation loss member in estimator class. The purpose of add the evaluation loss is to decouple the training loss with the evaluation loss in the fit_batch() and evaluate_batch() methods. Please refer to the issue for further information: https://github.com/apache/incubator-mxnet/issues/16879

### CODE REVIEW

File: python/mxnet/gluon/contrib/estimator/estimator.py [hunk 1]
```diff
@@ -64,13 +64,17 @@ def __init__(self, net,
...
-                 context=None):
+                 context=None,
+                 evaluation_loss=None):
 
         self.net = net
         self.loss = self._check_loss(loss)
         self._train_metrics = _check_metrics(metrics)
         self._add_default_training_metrics()
         self._add_validation_metrics()
+        self.evaluation_loss = None
```
Comment (leezu) [inline python/mxnet/gluon/contrib/estimator/estimator.py]: How about changing this line to `self.evaluation_loss = self.loss`. Then you can remove the if else clause in the `def evaluate_batch`

Comment (liuzh91) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: If users need customized evaluation loss in the base estimator, they cannot do that any more.

Comment (leezu) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: Why not? The semantics is simpler: `self.evaluation_loss` always refers to the loss used during evaluation. The proposed change is to simply initialize `self.evaluation_loss` with `self.loss` if the user does not specify `evaluation_loss` explicitly.

Comment (liuzh91) [reply python/mxnet/gluon/contrib/estimator/estimator.py]: Yes, you are right.

**Comment by leezu** [comment]
Could you also add a test case to https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_gluon_estimator.py to prevent future regressions of this feature

File: python/mxnet/gluon/contrib/estimator/estimator.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ class Estimator(object):
         Trainer to apply optimizer on network parameters.
     context : Context or list of Context
         Device(s) to run the training on.
+    evaluation_loss: gluon.loss.loss
+        Loss (objective) function to calculate during evaluation.
```
Comment (leezu) [inline python/mxnet/gluon/contrib/estimator/estimator.py]: It's not clear that `evaluation_loss` will use `loss` if it is None. Could you improve the description here?
